### DISCUSSION TITLE
How to resolve url for current language?

### DISCUSSION THREAD

**Discussion by cawa-93** [lumeland/lume#462] [Q&A] [answered]
Basically, I'm trying to combine `resolve_url` and `multilanguage` plugins.
I have [Multilanguage pages from a single file](https://lume.land/plugins/multilanguage/#multilanguage-pages-from-a-single-file) and want to resolve URL to it but for current language.

```ts
// _config.ts
site.use(multilanguage({
  languages: ["en", "uk"], // Available languages
  defaultLanguage: "uk", // The default language
}));
```

```tsx
// index.tsx

export const lang = ['uk', 'en']

export default () => (<a href="about.tsx"></a>) // always rendered as <a href="/about"></a> even if rendered for /en/
```

```tsx
// about.tsx

export const lang = ['uk', 'en']

export default () =>  (<a href="index.tsx"></a>) // always rendered as <a href="/"></a> even if rendered for /en/about/
```

**Comment by oscarotero** [comment, marked as answer]
Sorry, `resolve_url` doesn't have support for multilanguage pages. That's a good point, I'll think of a way to implement this.

**Reply by oscarotero** [reply]
I was thinking of adding support for queries after the filename. For example: `<a href="index.tsx(lang=en)">` returns the url of the page generated by `index.tsx` with the variable `lang=en`.

Do you think this can help you?

**Reply by cawa-93** [reply]
I guess Yes if I can just pass the target language.
```tsx
<a href={`index.tsx(lang=${currentLang})`}>
```

**Reply by oscarotero** [reply]
This has been landed in v1.18.4
